core/assets/currencies.go

```diff
@@ -232,3 +242,15 @@ func (*Eth) Symbol() string {
 func (e *Eth) ToInt() *big.Int {
 	return (*big.Int)(e)
 }
+
+// Float64 returns a float representation and an error if an approximation would
+// not fit within the float.
+func (e *Eth) Float64() (float64, error) {
```

Comment:
why error and not ok?  is the value expected to be discarded whenever there is one?

Reply:
Yeah, it will either be positive or negative infinity, which didn't seem like reasonable values.

Reply:
Maybe we should signal loss of accuracy?  If so, then the function probably belongs outside the 'assets' package to avoid misuse.

The documentation suggests the Inf is returned when the number is greater than or less than MaxFloat64 or SmallestNonZeroFloat64.

The thing is there will be loss of accuracy somewhere with a conversion between Int64 <-> and Float64, and this isn't accounted for here, which is should since we are dealing with money.

"Float64 returns the float64 -nearest- to x"

Reply:
This is really for monitoring, to get a quick sense of things, like e.g: the money is approaching zero. Not for accounting, as in: "the node has exactly N units" so I think the accuracy loss is ok.

Reply:
> This is really for monitoring, to get a quick sense of things, like e.g: the money is approaching zero. Not for accounting, as in: "the node has exactly N units" so I think the accuracy loss is ok.

I kind of mis- wrote but
"the function probably belongs outside the 'assets' package to avoid misuse."

It probably belongs elsewhere then, no in the assets package.  Expectations on this package should be pretty strict.  It will help avoid headaches in the future due to misapplication.